lib/operations/db_collection_ops.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+const applyWriteConcern = require('../utils').applyWriteConcern;
+const Collection = require('../collection');
+const executeCommand = require('./db_execute_commands').executeCommand;
+const f = require('util').format;
+const handleCallback = require('../utils').handleCallback;
+const MongoError = require('mongodb-core').MongoError;
+const ReadPreference = require('mongodb-core').ReadPreference;
+const shallowClone = require('../utils').shallowClone;
```

Comment:
please remove all uses of `shallowClone` with `Object.assign({}, obj1, obj2...})` while working through this code